src/api/api.go

```diff
@@ -74,118 +89,125 @@ func GroupsHandler(w http.ResponseWriter, r *http.Request) (interface{}, HttpErr
+		if err != nil {
+			return nil, HttpErrorf(http.StatusNotAcceptable, "Could not get ID from url: %v", err)
+		}
+		group, err := atcd.GetGroup(id)
+		if err != nil {
+			if IsNoSuchItem(err) {
+				return nil, HttpErrorf(http.StatusNotFound, "Invalid group")
+			}
+			return nil, HttpErrorf(http.StatusBadGateway, "Could not get group from daemon: %v", err)
 		}
-		return nil, HttpErrorf(http.StatusBadGateway, "Could not get group from daemon: %v", err)
+		return group, nil
+	case "OPTIONS":
+		return nil, nil
+	default:
+		return nil, InvalidMethod(r)
 	}
-	return group, nil
 }
 
 func GroupJoinHandler(w http.ResponseWriter, r *http.Request) (interface{}, HttpError) {
 	CORS(w, "POST")
 	atcd := GetAtcd(r)
-	if r.Method == "OPTIONS" {
+	switch r.Method {
+	case "POST":
+		id, err := strconv.ParseInt(mux.Vars(r)["id"], 10, 64)
+		if err != nil {
+			return nil, HttpErrorf(http.StatusNotAcceptable, "Could not get ID from url: %v", err)
+		}
+		req_info := &Token{}
+		if err := json.NewDecoder(r.Body).Decode(req_info); err != nil {
+			return nil, HttpErrorf(http.StatusNotAcceptable, "Could not parse json from request: %v", err)
+		}
+		member := GetClientAddr(r)
+		err = atcd.JoinGroup(id, member, req_info.Token)
+		if err != nil {
+			return nil, HttpErrorf(http.StatusBadGateway, "Could not join group: %v", err)
+		}
+		return MemberResponse{
+			Id:     id,
+			Member: member,
+		}, nil
+	case "OPTIONS":
 		return nil, nil
+	default:
+		return nil, InvalidMethod(r)
 	}
-	if r.Method != "POST" {
-		return nil, InvalidMethod
-	}
-	id, err := strconv.ParseInt(mux.Vars(r)["id"], 10, 64)
-	if err != nil {
-		return nil, HttpErrorf(http.StatusNotAcceptable, "Could not get ID from url: %v", err)
-	}
-	req_info := &Token{}
-	if err := json.NewDecoder(r.Body).Decode(req_info); err != nil {
-		return nil, HttpErrorf(http.StatusNotAcceptable, "Could not parse json from request: %v", err)
-	}
-	member := GetClientAddr(r)
-	err = atcd.JoinGroup(id, member, req_info.Token)
-	if err != nil {
-		return nil, HttpErrorf(http.StatusBadGateway, "Could not join group: %v", err)
-	}
-	return MemberResponse{
-		Id:     id,
-		Member: member,
-	}, nil
 }
 
 func GroupLeaveHandler(w http.ResponseWriter, r *http.Request) (interface{}, HttpError) {
 	CORS(w, "POST")
 	atcd := GetAtcd(r)
-	if r.Method == "OPTIONS" {
+	switch r.Method {
+	case "POST":
+		id, err := strconv.ParseInt(mux.Vars(r)["id"], 10, 64)
+		if err != nil {
+			return nil, HttpErrorf(http.StatusNotAcceptable, "Could not get ID from url: %v", err)
+		}
+		req_info := &Token{}
+		if err := json.NewDecoder(r.Body).Decode(req_info); err != nil {
+			return nil, HttpErrorf(http.StatusNotAcceptable, "Could not parse json from request: %v", err)
+		}
+		member := GetClientAddr(r)
+		err = atcd.LeaveGroup(id, member, req_info.Token)
+		if err != nil {
+			return nil, HttpErrorf(http.StatusBadGateway, "Could not join group: %v", err)
+		}
+		return MemberResponse{
+			Id:     id,
```

Comment:
IIRC, we should use ID  rather than Id due to https://github.com/golang/go/wiki/CodeReviewComments#initialisms but this is just nit and should not be addressed in this PR.